train.py

```diff
@@ -281,22 +420,23 @@ def evaluate(hps, generator, eval_loader, writer_eval):
     audio_dict = {}
     with torch.no_grad():
         for batch_idx, items in enumerate(eval_loader):
-            c, f0, spec, y, spk, _, uv,volume = items
-            g = spk[:1].cuda(0)
-            spec, y = spec[:1].cuda(0), y[:1].cuda(0)
-            c = c[:1].cuda(0)
-            f0 = f0[:1].cuda(0)
-            uv= uv[:1].cuda(0)
-            if volume!=None:
-                volume = volume[:1].cuda(0)
+            c, f0, spec, y, spk, _, uv, volume = items
+            g = spk[:1].to(f"{device}:0")
```

Comment:
Why not `to(device)` ?

Reply:
```python
device = (
    "cuda"
    if torch.cuda.is_available()
    else "mps"
    if torch.backends.mps.is_available()
    else "cpu"
)
```
Because `device` is defined as a string type like this `.to(device)` does not specify the destination and does not give the same result as the original code.